prowler/providers/aws/services/vpc/vpc_endpoint_connections_trust_boundaries/vpc_endpoint_connections_trust_boundaries.py

```diff
@@ -55,10 +55,14 @@ def execute(self):
                             break
 
                     else:
-                        if isinstance(statement["Principal"]["AWS"], str):
-                            principals = [statement["Principal"]["AWS"]]
+                        if "AWS" in statement["Principal"]:
+                            if isinstance(statement["Principal"]["AWS"], str):
+                                principals = [statement["Principal"]["AWS"]]
+                            else:
+                                principals = statement["Principal"]["AWS"]
                         else:
-                            principals = statement["Principal"]["AWS"]
+                            # If the principal is not an AWS principal, we don't need to check it since it could be a service or a federated principal
+                            principals = []
```

Comment:
Can this be moved to another place? I think some checks are using the same, please give it a look.